### ISSUE TITLE
FYI: Changing the renderer name is a breaking change for some use cases

### ISSUE THREAD

**Issue by aguynamedben** [electron/electron#22317]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 7.1.2
* **Operating System:**
  * macOS 10.15

### Expected Behavior

First off, I really appreciate Electron am so thankful for your work and support on it. I wanted to provide some feedback on a nuanced problem I ran into with keytar, macOS keychain, and the renderer process changing names with Electron 6.

The existing version of my app uses Electron 5.0.11, and uses keytar to store sensitive information in the macOS keychain. In this existing version, keytar sets values from within the renderer process, and, as macOS Keychain Access displays, token becomes available from the process "Command E Helper".

Now we're upgrading to Electron 7. The renderer name, due to Chromium changes, has changed from "Command E Helper" to "Command E Helper (Renderer)", as documented in the [Electron 6 release notes](https://www.electronjs.org/blog/electron-6-0#electron-helper-rendererapp-electron-helper-gpuapp-and-electron-helper-pluginapp).

When the new version of my app runs, any time the renderer gets a value from keytar, macOS doesn't trust "Command E Helper (Renderer)" to access it, because they keys were created with "Command E Helper".

Basically, I've inadvertently leaked Chromium/Electron internals (i.e. the name of the renderer process) to macOS via it's storage of permissions. Any ideas on a route out of this situation?

Questions:
* Is there anyway to tell Chromium to use a different piece of text for the renderer's process name? (i.e. so I can set the renderer process name back to "Command E Helper")

Feedback:
* The renderer process name changing should probably be listed in Breaking Changes on release notes. It's not a breaking change in itself, but if it's stored or relied on across versions, it can break things!

**Comment by MarshallOfSound** [comment]
The issue here isnt the process name changing, it's the bundle ID changing, you can probably hack around it by making the bundle ID of the renderer helper the same as the old helper bundle id, make sure you don't have two helpers with the same ID though.

If you want to solve it properly you need to call [SecItemUpdate](https://developer.apple.com/documentation/security/1393617-secitemupdate?language=objc) and add your new bundle ID to the access list for the password in the keychain.  It just re-add it to the keychain under your new bundle id and a new keyname.

Going to close this out as there isn't anything for us to do here

**Comment by tarahmarie** [comment]
Confirmed. Please reopen; I experienced this today. I denied all connection while confirming, and now even a delete and reinstall is causing issues. I'll sort more of this out tomorrow, but this is not a solved problem. 

<img width="688" alt="Screen Shot 2020-03-04 at 7 28 44 PM" src="https://user-images.githubusercontent.com/1920330/75951267-5c803980-5e60-11ea-8443-4b31f5460683.png">

**Comment by MarshallOfSound** [comment]
@tarahmarie that prompt / whatever issue you are experiencing is not this issue nor anything specific to Electron.  It looks like Signal changed the app bundle ID for the renderer helper and you have a tool (maybe littlesnitch) that noticed and is asking you to double check everything is OK.  If something has gone wrong with Signal or littlesnitch it is not an Electron issue.
